### ISSUE TITLE
Can't build with Haxe 3.2.1 and NekoVM 2.1.0 on Linux

### ISSUE THREAD

**Issue by FrancisBourre** [DoclerLabs/hexMachina#184]
_From @Sanva on December 8, 2016 12:54_

Hi!

I'm into trying `hexMachina`, but when I try to compile this *Hello World* example it simply doesn't work:

```
/hexMachina-HelloWorld $ haxe build/build.hxml -js bin/HelloWorld.js
/usr/share/haxe/std/neko/_std/EReg.hx:33: characters 11-61 : Unsupported escaped char '{'
<builtin>:1: character 0 : Called from
/usr/share/haxe/std/neko/_std/EReg.hx:22: lines 22-204 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/preprocess/Preprocessor.hx:9: characters 26-37 : Called from
<builtin>:1: character 0 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/preprocess/Preprocessor.hx:6: lines 6-59 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/preprocess/MacroPreprocessor.hx:21: characters 22-40 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/xml/XmlDSLParser.hx:41: characters 13-72 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/xml/XmlDSLParser.hx:29: characters 2-95 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/xml/XmlCompiler.hx:522: characters 22-99 : Called from
/home/v/haxelib/hexioc/0,10,0/src/hex/compiler/parser/xml/XmlCompiler.hx:588: characters 9-79 : Called from
./src/Main.hx:20: characters 2-56 : Called from
Aborted
```

Maybe an issue with current Haxe version or something like that? Maybe it won't compile on Linux?

_Copied from original issue: DoclerLabs/hexMachina-HelloWorld#2_

**Comment by FrancisBourre** [comment]
_From @vistyanpeter on December 9, 2016 8:38_

Hello Sanva,

I have made a quick test on Windows environment with the following versions and I can confirm that all the following options just build 
fine for me:

Haxe 3.2.1 with Neko 2.0.0
Haxe 3.2.1 with Neko 2.1.0
Haxe 3.4.0.rc1 with Neko 2.0.0
Haxe 3.3.0.rc1 with Neko 2.1.0 ( this is our current combination we use )

Unfortunately we have no option to try it on Linux ( We are Windows/OSX guys ), therefore your best shot might be to test your build with different Haxe and Neko versions together. If the issue still persist the problem might be on the Linux environment.

I hope it helps.
Good luck, and let us know how is that going...

Bests,
Peter

**Comment by FrancisBourre** [comment]
_From @st3veV on December 9, 2016 9:1_

I can reproduce it in the ubuntu subsystem on windows 10... I'll do some debugging.

**Comment by FrancisBourre** [comment]
_From @Sanva on December 9, 2016 10:33_

Hi again,

You may or may not have seen this already, but just in case, this may be related to https://github.com/HaxeFoundation/haxe/issues/4625#issuecomment-154029187

But the thing is that the solution proposed —the manual link `ln -s /usr/lib/libneko.so.0 /usr/lib/libneko.so`—does not have sense in my system, I only have `/usr/lib/libneko.so.2` and `/usr/lib/libneko.so.2.1.0`, with the first one being a link pointing to the later.

By the way, my Haxe 3.2.1 and NekoVM 2.1.0 are simply installed from official Haxe PPA.

**Comment by FrancisBourre** [comment]
_From @st3veV on December 9, 2016 11:49_

It's fixed now in the code and it should work everywhere. Very likely fix will get properly released with 0.12.0 version but you can use the `develop` branch of hexIoC

**Comment by Sanva** [comment]
Yes, I can compile it now —using, as you've suggested, the `develop` branch of hexIoC.
